src/coreclr/jit/llvm.cpp

```diff
@@ -1387,6 +1418,37 @@ void Llvm::buildCnsLng(GenTree* node)
     mapGenTreeToValue(node, _builder.getInt64(node->AsLngCon()->LngValue()));
 }
 
+llvm::Value* Llvm::buildFieldList(GenTreeFieldList* fieldList,
+                                  Type*             llvmType)
+{
+    if (!fieldList->TypeIs(TYP_STRUCT))
+    {
+        failFunctionCompilation();
+    }
```

Comment:
```suggestion
    assert(fieldList->TypeIs(TYP_STRUCT));
```

Reply:
Thanks, getting rid of `failFunctionCompilation` is the end goal after all.